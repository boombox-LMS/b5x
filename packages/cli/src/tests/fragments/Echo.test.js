const {
  exampleEchoMarkup,
  EchoDataSchema,
} = require("../../../dist/parser/componentBuilders/fragments/Echo");
const { RawFragmentSchema } = require("@b5x/types");

describe("The Echo fragment builds as expected", () => {
  const fragment = global.buildFragmentFromMarkupString(exampleEchoMarkup);
  const rawFragment = fragment.packageForApi();

  test("The parsed echo fragment matches the snapshot on file", () => {
    expect(rawFragment).toMatchSnapshot();
  });

  test("The parsed echo fragment is a valid RawFragment", () => {
    const validator = () => {
      RawFragmentSchema.parse(rawFragment);
    };
    expect(validator).not.toThrowError();
  });

  test("The parsed echo fragment uses the expected data schema", () => {
    const validator = () => {
      EchoDataSchema.parse(rawFragment.data);
    };
    expect(validator).not.toThrowError();
  });
});
