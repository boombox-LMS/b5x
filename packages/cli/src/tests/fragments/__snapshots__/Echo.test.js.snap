// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Echo fragment builds as expected The parsed echo fragment matches the snapshot on file 1`] = `
Object {
  "childUris": Array [],
  "contentType": "Echo",
  "contents": "<echo>my-user-inputted-value</echo>",
  "data": Object {
    "fragmentUrisByAlias": Object {
      "my-user-inputted-value": null,
    },
  },
  "dependencyUris": Array [],
  "digest": "f3cf3606e853b5424f5b83dc8dee6924",
  "displayConditions": Array [],
  "documentUri": "mock-document-uri",
  "isRequired": false,
  "isStateful": false,
  "uri": "test-fragment-uri",
}
`;
